examples/Rapid-product-visualization-for-client-presentations/main.bal

```diff
@@ -0,0 +1,106 @@
+
+    io:println("Product image generated successfully.");
+
+    // Save the generated product image in base64 format
+    string? b64GenImage = generationResponse.data[0].b64_json;
+
+    if (b64GenImage is string) {
+        io:println("Saving the generated product image...");
+        // Decode the base64 string to get image bytes
+        string|byte[]|io:ReadableByteChannel|mime:DecodeError genImageBytes = mime:base64Decode(b64GenImage.toBytes());
+        if (genImageBytes is byte[]) {
+            // Save the base image to a file
+            error? saveGenImageResult = io:fileWriteBytes("images/product_image.png", genImageBytes);
+
+            if (saveGenImageResult is error) {
+                io:println("Error writing the base image to a file: ", saveGenImageResult);
+                return;
+            } else {
+                io:println("Generated product image saved successfully as 'images/product_image.png'.");
+            }
+
+            io:println("Creating 3 variations of the generated product image...");
+
+            // Create 3 variations of the generated product image using the `images/variations` API
+            images:CreateImageVariationRequest variationRequest = {
+                image: {
+                    fileContent: genImageBytes,
+                    fileName: "product_image.png"
+                },
+                n: 3,
+                size: "1024x1024",
+                response_format: "b64_json"
+            };
+
+            images:ImagesResponse variationResponse = check openaiClient->/images/variations.post(variationRequest);
+
+            io:println("Variations generated successfully.");
+
+            // Save each variation to a file
+            foreach var i in 0 ..< variationResponse.data.length() {
+                string? b64VariationImage = variationResponse.data[i].b64_json;
+
+                if (b64VariationImage is string) {
+                    io:println("Saving variation ", i + 1, "...");
+                    // Decode the base64 string to get variation image bytes
+                    string|byte[]|io:ReadableByteChannel|mime:DecodeError variationImageBytes = mime:base64Decode(b64VariationImage.toBytes());
+                    if (variationImageBytes is byte[]) {
+                        // Save the variation image to a file
+                        string variationFileName = string `images/product_image_variation_${i + 1}.png`;
+                        error? saveVariationImageResult = io:fileWriteBytes(variationFileName, variationImageBytes);
+
+                        if (saveVariationImageResult is error) {
+                            io:println("Error writing variation ", i + 1, " to a file: ", saveVariationImageResult);
+                        } else {
+                            io:println("Variation ", i + 1, " saved successfully as '", variationFileName, "'.");
```

Comment:
we can use Ballerina string templates to improve the usability. Apply where necessary
```suggestion
                            io:println(string `Variation ${i + 1} saved successfully as '${variationFileName}'.`);
```